enterprise/server/githubapp/githubapp.go

```diff
@@ -1140,6 +1271,18 @@ func (a *GitHubApp) getGithubGraphQLClient(ctx context.Context) (*githubv4.Clien
 	return a.newAuthenticatedGraphQLClient(ctx, tu.GithubToken)
 }
 
+// IsTokenValid returns whether the oauth token is valid for the current app.
+func (a *GitHubApp) IsTokenValid(ctx context.Context, oauthToken string) bool {
+	// The Authorizations.Check API requires basic auth.
+	tp := github.BasicAuthTransport{
+		Username: a.oauth.ClientID,
+		Password: a.oauth.ClientSecret,
+	}
+	client := github.NewClient(tp.Client())
+	_, _, err := client.Authorizations.Check(ctx, a.oauth.ClientID, oauthToken)
+	return err == nil
```

Comment:
sanity check: will err be nil if the HTTP request succeeded but returned an auth error (401/403)? or will it be something like an HttpError

Reply:
It will return a 404 not found error in the `err` variable